src/components/Filters.tsx

```diff
@@ -7,92 +6,76 @@ export interface IFiltersProps<T> {
   onChangeFilter: (
     filterProperty: keyof T,
     checked: boolean,
-    isTruthyPicked: boolean
   ) => void;
-}
+};
 
-export function Filters<T>(props: IFiltersProps<T>) {
-  const { object, filters, onChangeFilter } = props;
+type radioButtonKeyType =
+  'sedu' |
+  'studioUnits' |
+  'oneBedroomUnits' |
+  'twoBedroomUnits' |
+  'threePlusBedroomUnits'
 
-  const labelTruthy = (
-    <>
-      {/* offered */}
-    </>
-  );
+type radioButtonUILabelType =
+  'Pod' |
+  'Studio' |
+  'One bedroom' |
+  'Two bedroom' |
+  'Three or more'
 
-  const labelFalsy = (
-    <>
-      is <b>falsy</b>
-    </>
-  );
-  const objectKeys = ["sedu", "studioUnits", "oneBedroomUnits", "twoBedroomUnits", "threePlusBedroomUnits"]
-  return (
-    <div className="p-1 my-2">
-      
-      {/* <label className="mt-3">Number of bedrooms:</label> */}
-      {/* Object is a class that is calling method keys on object (which is all the buildings) */}
-      {objectKeys.map((key) => {
-        if (key !== 'sedu' && 
-            key !== 'studioUnits' && 
-            key !== 'oneBedroomUnits' && 
-            key !== 'twoBedroomUnits' && 
-            key !== 'threePlusBedroomUnits') 
-            { return ('')}
-        
-        let styledKey = ''
+type radioButtonMapType = {
+  [key in radioButtonKeyType]: radioButtonUILabelType
+};
 
-        if (key === 'sedu') { 
-          styledKey = 'Pod'
-        } else if (key === 'studioUnits') {
-          styledKey = 'Studio'
-        } else if (key === 'oneBedroomUnits') {
-          styledKey = 'One bedroom'
-        } else if (key === 'twoBedroomUnits') {
-          styledKey = 'Two bedroom' 
-        } else if (key === 'threePlusBedroomUnits') {
-          styledKey = 'Three or more' 
-        }
+const radioButtonKeys: radioButtonKeyType[] = ['sedu', 'studioUnits', 'oneBedroomUnits', 'twoBedroomUnits', 'threePlusBedroomUnits']
 
-        const getRadioButton = (isTruthyPicked: boolean): ReactNode => {
-          const id = isTruthyPicked
-            ? `radio-defined-${key}`
-            : `radio-not-defined-${key}`;
-          const label = isTruthyPicked
-            ? labelTruthy
-            : labelFalsy;
+const radioButtonUILabels: radioButtonMapType = {
+  sedu: 'Pod',
+  studioUnits: 'Studio',
+  oneBedroomUnits: 'One bedroom',
+  twoBedroomUnits: 'Two bedroom',
+  threePlusBedroomUnits: 'Three or more',
+};
 
-          const getChecked = () => {
-            const x = filters.filter(x => x.property === key);
-            return x.length === 1 && x[0].isTruthyPicked === isTruthyPicked;
-          }
+export function Filters<T>(props: IFiltersProps<T>) {
+  const { filters, onChangeFilter } = props;
+
+  const getChecked = (radioButtonKey: radioButtonKeyType) => {
+    const x = filters.filter(x => x.property === radioButtonKey);
+    return x.length === 1;
+  };
+  
+  const getID = (radioButtonKey: radioButtonKeyType): radioButtonKeyType => {
+    return radioButtonKey;
+  };
```

Comment:
It seems like this function doesn't do anything?

Reply:
That's correct.
I got rid of it ed3a2c5!